From 0d2094ceb86a2e24c2d5b94a4b17ef6acc0abff7 Mon Sep 17 00:00:00 2001
From: Firelight Flagboy <firelight.flagboy@gmail.com>
Date: Sun, 18 Feb 2024 14:15:07 +0100
Subject: [PATCH] Fix `mountpoint_base_dir` not a `Path`

Fix `mountpoint_base_dir` not being converted to a `Path` in `load_config`.
---
 parsec/core/config.py | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/parsec/core/config.py b/parsec/core/config.py
index 1b9341326..491c019b9 100644
--- a/parsec/core/config.py
+++ b/parsec/core/config.py
@@ -215,15 +215,11 @@ def load_config(config_dir: Path | None = None, **extra_config: Any) -> CoreConf
         logger.warning("Ignoring invalid config", config_file=config_file, error=str(exc))
         data_conf = {}
 
-    try:
-        data_conf["data_base_dir"] = Path(data_conf["data_base_dir"])
-    except (KeyError, ValueError):
-        pass
-
-    try:
-        data_conf["prevent_sync_pattern_path"] = Path(data_conf["prevent_sync_pattern_path"])
-    except (KeyError, ValueError):
-        pass
+    for path_key in ("data_base_dir", "mountpoint_base_dir", "prevent_sync_pattern_path"):
+        try:
+            data_conf[path_key] = Path(data_conf[path_key])
+        except (KeyError, ValueError):
+            pass
 
     try:
         data_conf["disabled_workspaces"] = frozenset(
-- 
2.42.0

